src/test/java/org/benetech/servicenet/service/impl/ImportServiceConfidentialityImplIntTest.java

```diff
@@ -295,213 +333,159 @@ public void shouldNotUpdateServiceIfConfidential() {
-        Service service = generateExistingService();
-        Phone phone = new Phone().number(NEW_STRING).srvc(service);
+        Phone phone = new Phone().number(NEW_STRING);
         phone.setIsConfidential(true);
+        Service service = generateExistingService().phones(Set.of(phone));
 
-        int dbSize = phoneService.findAll().size();
-        importService.createOrUpdatePhonesForService(Set.of(phone), service, null);
+        importManager.createOrUpdateService(service, EXISTING_EXTERNAL_ID, PROVIDER,
+            new DataImportReport());
 
-        assertEquals(dbSize, phoneService.findAll().size());
+        assertEquals(0, phoneService.findAll().size());
     }
 
     @Test
     @Transactional
     public void shouldNotReplacePhonesIfConfidentialEvenIfServiceHasFewOfThemButNotThisOne() {
-        Service service = generateExistingService();
-        Phone otherPhone = new Phone().number(OTHER_STRING).srvc(service);
-        em.persist(otherPhone);
-        em.flush();
-        em.refresh(service);
+        Phone otherPhone = new Phone().number(OTHER_STRING);
+        Service service = generateExistingService().phones(Set.of(otherPhone));
+        importManager.createOrUpdateService(service, EXISTING_EXTERNAL_ID, PROVIDER, new DataImportReport());
 
-        Phone newPhone = new Phone().number(NEW_STRING).srvc(service);
+        Phone newPhone = new Phone().number(NEW_STRING);
         newPhone.setIsConfidential(true);
+        service.setPhones(Set.of(newPhone));
 
-        importService.createOrUpdatePhonesForService(Set.of(newPhone), service, null);
+        importManager.createOrUpdateService(service, EXISTING_EXTERNAL_ID, PROVIDER, new DataImportReport());
 
         assertEquals(0, phoneService.findAll().size());
     }
 
     @Test
     @Transactional
     public void shouldNotReplacePhonesIfConfidential() {
-        Service service = generateExistingService();
-
-        Phone phoneToBeUpdated = new Phone().number(EXISTING_STRING).srvc(service);
-        em.persist(phoneToBeUpdated);
-
-        Phone otherPhone = new Phone().number(OTHER_STRING).srvc(service);
-        em.persist(otherPhone);
-        em.flush();
-        em.refresh(service);
+        Phone phoneToBeUpdated = new Phone().number(EXISTING_STRING);
+        Phone otherPhone = new Phone().number(OTHER_STRING);
+        otherPhone.setIsConfidential(true);
+        Service service = generateExistingService().phones(Set.of(phoneToBeUpdated, otherPhone));
 
-        Phone newPhone = new Phone().number(NEW_STRING).srvc(service);
-        newPhone.setIsConfidential(true);
-
-        importService.createOrUpdatePhonesForService(Set.of(newPhone), service, null);
+        importManager.createOrUpdateService(service, EXISTING_EXTERNAL_ID, PROVIDER, new DataImportReport());
 
-        assertEquals(0, phoneService.findAll().size());
+        assertEquals(1, phoneService.findAll().size());
+        assertEquals(EXISTING_STRING, phoneService.findAll().get(0).getNumber());
```

Comment:
In my opinion, this is misleading because the phone is outdated.

Reply:
same here. I agree, but we probably have to create a separate a task for it, as it's working the same way as before changing the ImportService.

Reply:
You're right in this case, it shouldn't happened.